rust/libnewsboat/src/filterparser.rs

```diff
@@ -0,0 +1,287 @@
+extern crate nom;
+
+use std::io::{self, Read, BufRead};
+use nom::*;
+use nom::types::CompleteStr;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Operation {
+    Equals,
+    NotEquals,
+    RegEquals,
+    NotRegEquals,
+    LessThan,
+    GreaterThan,
+    LessThanOrEquals,
+    GreaterThanOrEquals,
+    Between,
+    Contains,
+    NotContains,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Value {
+    Str(String),
+    Int(i32),
+    Range(i32, i32)
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Comparison {
+    pub attribute: String,
+    pub op: Operation,
+    pub value: Value
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Expression {
+    And(Box<Expression>, Box<Expression>),
+    Or(Box<Expression>, Box<Expression>),
+    Comparison(Comparison)
+}
+
+named!(operators<CompleteStr, Operation>,
+    do_parse!(
+        tag: alt!(
+            map!(tag!("=="), |_| Operation::Equals) |
+            map!(tag!("!="), |_| Operation::NotEquals) |
+            map!(tag!("=~"), |_| Operation::RegEquals) |
+            map!(tag!("=") , |_| Operation::Equals) |
+            map!(tag!("!~"), |_| Operation::NotRegEquals) |
+            map!(tag!("<="), |_| Operation::LessThanOrEquals) |
+            map!(tag!(">="), |_| Operation::GreaterThanOrEquals) |
+            map!(tag!("<") , |_| Operation::LessThan) |
+            map!(tag!(">") , |_| Operation::GreaterThan) |
+            map!(tag!("between"), |_| Operation::Between) |
+            map!(tag!("#"), |_| Operation::Contains) |
+            map!(tag!("!#"), |_| Operation::NotContains)
+        ) >>
+        (tag)
+    )
+);
+
+fn quoted_string(input: CompleteStr) -> IResult<CompleteStr, Value>{
+    do_parse!(input,
+        tag!("\"")
+        >> value: escaped_transform!(
+                    is_not!("\\\""),
+                    '\\',
+                    alt!(
+                        tag!("\\") => { |_| &"\\"[..] } |
+                        tag!("\"") => { |_| &"\""[..] }
+                    )
+            )
+        >> tag!("\"")
+        >> (value)
+    ).map(|result| {
+        (result.0, Value::Str(result.1))
+    })
+}
+
+fn number(input: CompleteStr) -> IResult<CompleteStr, Value> {
+    do_parse!(input,
+        num: recognize!(
```

Comment:
Is it possible to do `num: number` like `range` below https://github.com/newsboat/newsboat/pull/701/files#diff-d5d63191785846ad8b3d3e5c6a00cd08R100?

Reply:
This is inside the `number` function, where that filter is being defined, so it's not possible :)

Reply:
Oh